rusoto/services/sts/src/custom/credential.rs

```diff
@@ -270,6 +275,58 @@ impl StsAssumeRoleSessionCredentialsProvider {
         }
     }
 
+    /// Creates a new `StsAssumeRoleSessionCredentialsProvider` with the given [ProfileProvider].
+    ///
+    /// * `profile_provider` - [ProfileProvider] for the profile containing session information.
+    pub fn with_profile_provider(
+        profile_provider: ProfileProvider,
+    ) -> Result<StsAssumeRoleSessionCredentialsProvider, CredentialsError> {
+        let mut profile_provider = profile_provider.clone();
+
+        let properties = profile_provider.profile_properties()?;
+
+        // in case the profile has a source_profile used to assume the role, that will be our
+        // credentials provider for our client
+        if let Some(source) = properties.get(SOURCE_PROFILE) {
+            profile_provider.set_profile(source);
+        };
+
+        let region = profile_provider
+            .region_from_profile()?
+            .map_or_else(|| Ok(Region::default()), |r| Region::from_str(&r))
+            .map_err(|e| CredentialsError::new(e))?;
+
+        let sts_client = StsClient::new_with(
+            HttpClient::new().expect("failed to create request dispatcher"),
+            profile_provider,
+            region,
+        );
+
+        let role_arn = properties
+            .get(ROLE_ARN)
+            .cloned()
+            .ok_or(CredentialsError::new(
+                "Profile is missing mandatory role_arn entry",
+            ))?;
+        let session_name =
+            properties
+                .get(ROLE_SESSION_NAME)
+                .cloned()
+                .ok_or(CredentialsError::new(
+                    "Profile is missing mandatory role_session_name entry",
+                ))?;
```

Comment:
I just realised, I never had a `role_session_name` in my .aws/credentials file, is it possible to make it optional?

If I look into the documentation: https://docs.aws.amazon.com/credref/latest/refdocs/setting-global-role_session_name.html, it has a default value:

> Default value: An optional parameter. If you don't provide this value, a session name is generated automatically if the profile assumes a role

Reply:
I could. However that would change the behaviour of `StsAssumeRoleSessionCredentialsProvider` (as defined in rusoto) cf.  https://github.com/rusoto/rusoto/blob/master/rusoto/services/sts/src/custom/credential.rs#L254 which would be a breaking change :( 

I didn't have this defined in my credentials either :/

Maybe we could generate this if not available, but I guess this would be an undesirable behaviour.

Reply:
You're right!

I'll open an issue to track this problem not directly related to this PR. Thanks